pkg/trace/metrics/timing/timing.go

```diff
@@ -0,0 +1,106 @@
+package timing
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/atomic"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+)
+
+// Set represents a set of metrics that can be used for timing. Use NewSet
+// to create a set. It is safe for concurrent use.
+type Set struct {
+	c   map[string]*counter // maps names to their aggregates
+	ctx context.Context     // used for cancelling when auto-reporting
+}
+
+// NewSet returns a new Set containing the given metric names. The context is optional
+// and can be used as cancellation for auto-reporting sets.
+func NewSet(ctx context.Context, names ...string) *Set {
+	set := Set{
+		c:   make(map[string]*counter, len(names)),
+		ctx: ctx,
+	}
+	for _, name := range names {
+		set.c[name] = newCounter(name)
+	}
+	return &set
+}
+
+// Autoreport enables autoreporting of the Set at the given interval.
+func (s *Set) Autoreport(interval time.Duration) {
+	go func() {
+		tick := time.NewTicker(interval)
+		defer tick.Stop()
+		for {
+			select {
+			case <-tick.C:
+				s.Report()
+			case <-s.ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+// Measure measures the time passed since, for the given name. It aggregates on each
+// subsequent call until Report is called.
+func (s *Set) Measure(name string, since time.Time) {
+	ms := float64(time.Since(since) / time.Millisecond)
```

Comment:
> ```go
> defer timing.Since("my.metric", time.Now())
> ```

Yeah that's pretty nice. Also it's different enough so it's unlikely to confuse anyone. 👍